src/pages/publication/FilesAndLicensePanel.tsx

```diff
@@ -40,9 +42,23 @@ interface FilesAndLicensePanelProps {
 
 const FilesAndLicensePanel: React.FC<FilesAndLicensePanelProps> = ({ goToNextTab, uppy }) => {
   const { t } = useTranslation('publication');
-  const { values }: FormikProps<FormikPublication> = useFormikContext();
+  const { values, setFieldTouched }: FormikProps<FormikPublication> = useFormikContext();
   const [isLicenseModalOpen, setIsLicenseModalOpen] = useState(false);
 
+  const valuesRef = useRef(values);
+  useEffect(() => {
+    valuesRef.current = values;
+  }, [values]);
```

Comment:
har ikke dette akkurat samme funksjonalitet som useCallback? Noen grunn til at du bruker useRef + useEffect isteden her?

Reply:
useCallback vil opprette en ny funksjon  hver gang values endres, som igjen vil trigge useEffecten som setter at alle felter er touched

Vi må ha tilgang til alle opplastede filer for å kunne sette alle felter til touched. Bruker derfor useRef slik at vi kan bruke oppdatert verdi av values uten at hver endring i values trigger at useEffecten som setter touched kjører

(hvis jeg har forstått det riktig :p )

Reply:
Jeg bare skjønner ikke helt poenget med å bruke Ref, når du uansett oppdaterer current-verdien hver gang values endrer seg?

Reply:
det siste eksemplet bruker ikke values (her må vi vite hvor mange filer som er lastet opp for å finne alle felter til hver enkelt fil). Hvis values hadde vært en dependency til useCallback så ville den funksjonen blitt oppdatert hver gang man oppdaterte verdien i ett felt. Dette ville igjen trigget useEffect, og alle felter hadde blitt touched med en gang man oppdaterer en verdi.

Men hvis vi ikke hadde vært avhengig av values her så hadde det nok vært det samme ja

Reply:
ahhhh, så du bruker valuesRef for å sette verdien _etter_ et felt er endret på! da er jeg med 💡 (tror jeg) 😝